### PR TITLE
Update IsNonCodeOnly to support contrib…

### PR DESCRIPTION
**Pull request by vdemeester**
… and refactor a bit plus some tests. 🐙

I wonder about `OnlyFreeBSD` and `OnlyWindows` ones, are they still in use ? ('cause the code seemed a little bit wrong — `HasPrefix` instead of `HasSuffix`)

The `isContrib` list is not complete, and I really wish we could ignore the folder all together, but it's used during the build, I'll have to see what to do about that :wink:.

/cc @jfrazelle @calavera :angel: 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: github/pull_request.go [hunk 1]
```diff
@@ -153,9 +152,9 @@ func (p *pullRequestContent) OnlyFreebsd() bool {
 	var linux bool
 
 	for _, f := range p.files {
```
Comment (vdemeester) [inline github/pull_request.go]: I was thinking the code *should* be something like:

```go
for _, f := range p.files {
    if !isFreeBSD(f.FileName) {
        return false
    }
}
return true
```

And almost the same for `OnlyWindows()` below..

**Comment by calavera** [comment]
LGTM

**Comment by jfrazelle** [comment]
LGTM

File: github/pull_request.go [hunk 2]
```diff
@@ -153,9 +152,9 @@ func (p *pullRequestContent) OnlyFreebsd() bool {
...
-		if strings.HasPrefix(f.FileName, "_freebsd.go") {
+		if strings.HasSuffix(f.FileName, "_freebsd.go") {
```
Comment (thaJeztah) [inline github/pull_request.go]: wah?

hahaha :smile:

**Pull request merged**